### PR TITLE
properly document guaruntee provided by zebra-network

### LINKED ISSUE #710: Document guarantees provided by zebra-network
Yep, this should probably be better documented because @teor2345 was surprised too in a different context (in that instance it was that ZN guarantees that if you request blocks by hash you'll either get those blocks or you'll get an error, there's no way to get a different unrelated block).

_Originally posted by @hdevalence in https://github.com/ZcashFoundation/zebra/pull/707/review_comment/create_

### PR DESCRIPTION
**Pull request by yaahc**
closes https://github.com/ZcashFoundation/zebra/issues/710

### CODE REVIEW

**Review by teor2345 — DISMISSED** [review]
Looks good to me.

Question: Did we want to add similar text to the relevant doc as well?

**Comment by yaahc** [comment]
@teor2345 not sure what you mean by "the relevant doc"

File: zebra-network/src/lib.rs [hunk 1]
```diff
@@ -12,6 +12,10 @@
 //! [`tower::Service`] representing "the network" which load-balances
 //! outbound [`Request`]s over available peers.
 //!
+//! Unlike the underlying legacy network protocol the provided
+//! [`tower::Service`] guaruntees that each `Request` future will resolve to
```
Comment (dconnolly) [inline zebra-network/src/lib.rs]:
```suggestion
//! [`tower::Service`] guarantees that each `Request` future will resolve to
```

**Review by dconnolly — CHANGES_REQUESTED** [review]
One typo then lgtm!

**Review by dconnolly — APPROVED** [review]

File: zebra-network/src/lib.rs (hunk 1 above)
Comment (yaahc) [reply zebra-network/src/lib.rs]: I swear to god this word is my arch nemesis. When I wrote this I was like, "did I spell it wrong this time? naw, looks fine. :smiley:"

NOPE

**Pull request merged**